device/Program.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Microsoft.Azure.Amqp.Framing;
+using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Shared;
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace device
+{
+    class Program
+    {
+        static bool sendModelAtConnection = true;
+        static bool sendModelAsProperties = true;
+        static string cs = System.Environment.GetEnvironmentVariable("DEVICE_CS");
+        static string model = File.ReadAllText(@"..\..\..\deviceModel.json");
+
+        static async Task Main(string[] args)
+        {
+            string selfDescribingDtmi = "dtmi:azure:common:SelfDescribing;1";
+            string targetModelId = "dtmi:com:example:myDevice;1";
+            string targetModelhash = common.Hash.GetHashString(model);
+
+            string modelId = selfDescribingDtmi;
+
+            if (sendModelAtConnection)
```

Comment:
I'm not sure why we have this boolean. We send it at connection no matter what on line 33.

The difference, it seems, is that the model Id is based on a standard one, or a self-describing one. Is that right?

Reply:
Wait, no the difference is whether you have tmhash and tmid appended?

Reply:
this prototype allows to specify the targetHash and the targetId as optional, using querystring or twin properties. The boolean(s) help to validate different options.

Reply:
Your boolean name is misleading then.